emission/net/ext_service/habitica/proxy.py

```diff
@@ -14,29 +16,47 @@
-  u = requests.post(register_url, json=user_request)
-  # Bail out if we get an error
-  u.raise_for_status()
-  user_dict = json.loads(u.text)
-  logging.debug("parsed json from habitica has keys = %s" % user_dict.keys())
-
-  #Since we are randomly generating the password, we store it in case users 
-  #want to access their Habitica account from the browser
-  #Need to create a way from them to retrieve username/password
-  #metrics_data is used to calculate points based on km biked/walked
-  #last_timestamp is the last time the user got points, and bike/walk_count are the leftover km
-  habitica_user_table = edb.get_habitica_db()
-  habitica_user_table.insert({'user_id': our_uuid, 
-    'metrics_data': {'last_timestamp': arrow.utcnow().timestamp, 'bike_count': 0, 'walk_count': 0},
-    'habitica_username': username, 
-    'habitica_password': password, 
-    'habitica_id': user_dict['data']['_id'], 
-    'habitica_token': user_dict['data']['apiToken']})
+  user_dict = {}
+  #try to login and see if user is already in Habitica
+  try:
+    login_url = url + '/api/v3/user/auth/local/login'
+    user_request = {'username': username,'email': email,'password': password}
+    logging.debug("About to login %s"% user_request)
+    u = requests.post(login_url, json=user_request)
+    user_dict = json.loads(u.text)
+    logging.debug("parsed json from habitica has keys = %s" % user_dict.keys())
+
+  #if u fails, then user is not in Habitica, so needs to create an account
+  except urllib2.HTTPError:
+    register_url = url + '/api/v3/user/auth/local/register'
+    user_request = {'username': username,'email': email,'password': password,'confirmPassword': password}
+    logging.debug("About to register %s"% user_request)
+    u = requests.post(register_url, json=user_request)
+    # Bail out if we get an error
+    u.raise_for_status()
+    user_dict = json.loads(u.text)
+    logging.debug("parsed json from habitica has keys = %s" % user_dict.keys())
+
+  #if user is not in e-mission db, create it
+  if edb.get_habitica_db().find({'user_id': our_uuid}).count() == 0:
+    #Since we are randomly generating the password, we store it in case users 
+    #want to access their Habitica account from the browser
+    #Need to create a way from them to retrieve username/password
+    #metrics_data is used to calculate points based on km biked/walked
+    #last_timestamp is the last time the user got points, and bike/walk_count are the leftover km
+    habitica_user_table = edb.get_habitica_db()
+    habitica_user_table.insert({'user_id': our_uuid, 
+      'metrics_data': {'last_timestamp': arrow.utcnow().timestamp, 'bike_count': 0, 'walk_count': 0},
+      'habitica_username': username, 
+      'habitica_password': password, 
+      'habitica_id': user_dict['data']['_id'], 
+      'habitica_token': user_dict['data']['apiToken'],
+      'habitica_group_id': ""})
+  
+  #Do initial setup and append GroupId to the user
+  group_id = assist.user_setup(our_uuid)
```

Comment:
this is outside the emission db check, right? Is it idempotent? 
Recall that the user may call `habiticaRegister` multiple times and our goal is to support that.

Right now, I think that if the user calls `habiticaRegister` a second time, she will be randomly re-assigned to a different group.